config/load.go

```diff
@@ -61,10 +61,21 @@ func Load(data []byte) (map[string]interface{}, error) {
 		return nil, fmt.Errorf("Expected 'config' property to be an object")
 	}
 
+	// Ensure that we have a simple JSON compatible structure
+	if err := jsonCompatTypes(result); err != nil {
+		panic(fmt.Sprintf("YAML loaded wrong types, error: %s", err))
+	}
+
 	// Apply transforms
 	if _, ok := c["transforms"]; ok {
```

Comment:
instead of ignoring the result, shouldn't we save it in a variable, since we will need this in line 72?